include/albatross/src/core/dataset.hpp

```diff
@@ -157,8 +157,8 @@ inline auto concatenate_datasets(const RegressionDataset<X> &x,
 
 template <typename X, typename std::enable_if_t<
                           albatross::is_streamable<X>::value, int> = 0>
-std::ostream &operator<<(std::ostream &os,
-                         const albatross::RegressionDataset<X> &dataset) {
+inline std::ostream &
```

Comment:
What's the motivation behind using `inline` here? I don't know actual guidance, but its used when the time of cumulative function calls costs more than the added compile size... so its best for small functions called from a few place many times when CPU bound.

Reply:
Actually for templated functions like this it's mandatory, without it the linker will find multiple definitions, it worked fine before because I was only ever printing a dataset of a particular type once, but if I had printed it in multiple files I would have hit a multiple definition error in the linker.